test/skimNtuple_HHbtag.cpp

```diff
@@ -1562,23 +1487,60 @@ int main (int argc, char** argv)
 		  theSmallTree.m_nRealTaus = nRealTaus;                     // -1: data; > 0: # real taus in MC
 		}
 
-	  const TLorentzVector tlv_firstLepton (theBigTree.daughters_px->at (firstDaughterIndex),
-											theBigTree.daughters_py->at (firstDaughterIndex),
-											theBigTree.daughters_pz->at (firstDaughterIndex),
-											theBigTree.daughters_e->at (firstDaughterIndex)
-											);
+	  TLorentzVector tlv_firstLepton(theBigTree.daughters_px->at(firstDaughterIndex),
+									 theBigTree.daughters_py->at(firstDaughterIndex),
+									 theBigTree.daughters_pz->at(firstDaughterIndex),
+									 theBigTree.daughters_e->at(firstDaughterIndex));
+	  TLorentzVector tlv_secondLepton(theBigTree.daughters_px->at(secondDaughterIndex),
+									  theBigTree.daughters_py->at(secondDaughterIndex),
+									  theBigTree.daughters_pz->at(secondDaughterIndex),
+									  theBigTree.daughters_e->at(secondDaughterIndex));
+
+	  // electron energy scales and smears
+	  double uncorrEn1 = tlv_firstLepton.E();
+	  double uncorrEn2 = tlv_secondLepton.E();
+	  if (isMC) {
+		double corr1 = theBigTree.daughters_ecalTrkEnergyPostCorr->at(firstDaughterIndex);
+		double corr2 = theBigTree.daughters_ecalTrkEnergyPostCorr->at(secondDaughterIndex);
+		if (corr1 == -999. or corr2 == -999.) {
+		  throw std::runtime_error("ERROR: unchanged electron energy scale!");
+		}
```

Comment:
These checks will have to be done within the if(oph.isElectron(type)), right? If I understand correctly this would now throw an exception as soon as a daughter is not an electron (independent of if you're trying to apply this correction or not)